src/trio/_core/_run.py

```diff
@@ -1210,7 +1260,14 @@ def _child_finished(
     ) -> None:
         self._children.remove(task)
         if isinstance(outcome, Error):
-            self._add_exc(outcome.error)
+            self._add_exc(
+                outcome.error,
+                CancelReason(
+                    source="nursery",
+                    source_task=repr(task),
```

Comment:
If we unconditionally set `source_task` in `.cancel()` we can avoid this duplication!

Reply:
I'm not sure what the duplication is here that you wanted to avoid?

Reply:
I misread this somehow, sorry.